pkg/chaincode/mock/mock_client.go

```diff
@@ -0,0 +1,92 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package mock
+
+import (
+	"context"
+	"sync"
+
+	"github.com/hyperledger/fabric-protos-go-apiv2/common"
+	"github.com/hyperledger/fabric-protos-go-apiv2/orderer"
+	"github.com/hyperledger/fabric-protos-go-apiv2/peer"
+	"google.golang.org/grpc"
+)
+
+type EndorserClient struct {
```

Comment:
This looks like generated code. Do we really need to check it in or can it be generated like the other mocks used in this package? Right now, this would be part of the module's public API, which it really shouldn't be.

See here for an example of a generated mock:

https://github.com/hyperledger/fabric-admin-sdk/blob/a2ad1917cce51af5a5be83a9321af881d752a42e/pkg/chaincode/install_test.go#L27-L28

Note the code in `install_test.go` that demonstrates how you can use a generated mock for the gRPC connection that sits underneath the endorser client, so you don't actually need to mock the EndorserClient itself. If you run `make generate` you can then use the same mock in your test code.

Reply:
will have a try